src/libexpr/primops.cc

```diff
@@ -1979,6 +2010,15 @@ void EvalState::createBaseEnv()
         for (auto & primOp : *RegisterPrimOp::primOps)
             addPrimOp(std::get<0>(primOp), std::get<1>(primOp), std::get<2>(primOp));
 
+#if __APPLE__
+    // Xcode integration
+    mkString(v, lookupXcodeVersion());
+    addConstant("__xcodeVersion", v);
+
+    mkString(v, lookupSDKRoot());
+    addConstant("__xcodeSDKRoot", v);
```

Comment:
This is not lazy, so it will run two child processes at startup time, and throw an exception if Xcode is not installed. I think the easiest way to make this lazy is to turn it into a primop with one dummy argument, and then use `mkApp` to pass the dummy argument.